CIPs/CIP-79/cip-79.md

```diff
@@ -0,0 +1,270 @@
+
+Now the content of the document should consist of a JSON object with one property *publicKeys*. The value of this property should be an object where the value is a [multicodec](https://github.com/multiformats/multicodec/) *base58btc* encoded public key, the key for any given key should be the last *15* characters of the encoded public key.
+
+The 3ID DID method supports any type of public key that can be encoded using multicodec which can easily be extended to support quantum resistant signature and encryption schemes in the future.
+
+#### Example
+
+Below you can see an example of how to create a 3ID using the Ceramic javascript api.
+
+```js
+const doc = await ceramic.createDocument({
+  metadata: {
+    controllers: ['did:key:zQ3shQNcackrTByiYaPGso1Nt7b6r1gSMg4XXBmavzvTMqX1h'], // secp256k1 did:key
+    family: '3id'
+  },
+  content: {
+    publicKeys: {
+      "XQCT2xJHsdY6iJH": "z6LSqKWh3XQ7AfsJuE2KR23cozEut8D5CXQCT2xJHsdY6iJH",  // secp256k1 public key
+      "yh27jTt7Ny2Pwdy": "zQ3shrMGEKAjUTMmvDkcZ7Y3x9XnVjTH3myh27jTt7Ny2Pwdy", // x25519 public key
+    }
+  },
+  deterministic: true
+})
+
+const didString = `did:3:${doc.id}`
+```
+
+### Read (Resolve)
+
+Resolving a 3ID is quite straight forward. It is done by taking the DocID from the method specific identifier, looking up the referred document using Ceramic, and converting the content of the Ceramic document into a DID document.
+
+#### Loading 3IDv1 document
+
+To load a 3IDv1 document simply take the DocID from the method specific identifier and load the document from ceramic.
+
+#### Loading 3IDv0 document
+
+To load a 3IDv0 document, first take the CID from the method specific identifier and load the corresponding object from the IPFS network. The resolved object will contain a `publicKey` property which has an array of public key objects (see [Appendix A](#appendix-a)).
+
+Now find the key with an id that ends in `#signingKey` and convert it to a multicodec encoded key (in compressed format), then use it as a `did:key` and look up the ceramic document which has this key as the *controller* and *family* set to `3id`, using the *deterministic* flag. Below you can see the code to do this with the javascript Ceramic api.
+
+
+
+```js
+const doc = await ceramic.createDocument({
+  metadata: {
+    controllers: [didKey], // secp256k1 did:key
+    family: '3id'
+  },
+  deterministic: true
+})
+```
+
+#### Converting the Ceramic document to the DID document
+
+Once we have the Ceramic document loaded, load the latest *AnchorCommit* of the document (the latest commit that was anchored to a blockchain). Then get the content of the document which will look something like this:
```

Comment:
Here we say just to load the latest `AnchorCommit` but below we reference both the earliest and latest `AnchorCommit`

Reply:
Well yeah. Once you have loaded the latest `AnchorCommit` you should be able to easily get the data from the earliest one?

Reply:
its just a weird asymmetry.  It's kind of confusing that we mention loading the `AnchorCommit` here at all when we don't actually do anything with it until several steps later.  Maybe just remove the reference from this section entirely and just make sure to define the term `AnchorCommit` later when you actually need it for the first time?

Reply:
We do do something with it since we use the content from the latest anchorCommit. Clarified this a bit now.